src/mscorlib/shared/System/String.Comparison.cs

```diff
@@ -828,9 +835,16 @@ public static bool Equals(string a, string b, StringComparison comparisonType)
                     if (a.Length != b.Length)
                         return false;
 #if CORECLR
-                    // If both strings are ASCII strings, we can take the fast path.
-                    if (a.IsAscii() && b.IsAscii())
+                    var aIsAscii = a.IsAscii();
+                    var bIsAscii = b.IsAscii();
+                    if (aIsAscii != bIsAscii)
+                    {
+                        // If one string is ASCII and other is not; then they are not equal
+                        return false;
+                    }
+                    else if (aIsAscii == true && aIsAscii == bIsAscii)
```

Comment:
Again, at this moment we know that `aIsAscii == bIsAscii`.

Reply:
Changed